src/main/kotlin/no/nav/familie/tilbake/behandling/BehandlingService.kt

```diff
@@ -169,17 +191,53 @@ class BehandlingService(private val behandlingRepository: BehandlingRepository,
                       fagsystem = fagsystem)
     }
 
-    private fun kanHenleggeBehandling(behandling: Behandling): Boolean {
+    private fun kanHenleggeBehandling(behandling: Behandling,
+                                      behandlingsresultatstype: Behandlingsresultatstype? = null): Boolean {
         var kanHenlegges = true
-        if (TILBAKEKREVING == behandling.type) {
+        if (Behandlingsresultatstype.HENLAGT_KRAVGRUNNLAG_NULLSTILT == behandlingsresultatstype) {
+            kanHenlegges = true
+        } else if (TILBAKEKREVING == behandling.type) {
             kanHenlegges = !behandling.erAvsluttet() && (!behandling.manueltOpprettet &&
                                                          behandling.opprettetTidspunkt < LocalDate.now()
                                                                  .atStartOfDay()
                                                                  .minusDays(OPPRETTELSE_DAGER_BEGRENSNING))
+                           && !kravgrunnlagRepository.existsByBehandlingIdAndAktivTrue(behandling.id)
         }
         return kanHenlegges
     }
 
+    private fun kanSendeHenleggelsesbrev(behandling: Behandling, behandlingsresultatstype: Behandlingsresultatstype): Boolean {
+        when (behandling.type) {
+            TILBAKEKREVING -> {
+                if (brevsporingRepository.existsByBehandlingIdAndBrevtypeIn(behandling.id,
+                                                                            setOf(Brevtype.VARSEL, Brevtype.KORRIGERT_VARSEL))) {
+                    return true
+                }
+            }
+            REVURDERING_TILBAKEKREVING -> {
+                if (Behandlingsresultatstype.HENLAGT_FEILOPPRETTET_MED_BREV == behandlingsresultatstype) {
+                    return true
+                }
+            }
+        }
+        return false
+    }
+
+    private fun finnBehandling(behandlingId: UUID): Behandling {
+        return behandlingRepository.findByIdOrNull(behandlingId)
+               ?: throw Feil(message = "Behandling finnes ikke for behandlingId=$behandlingId",
+                             frontendFeilmelding = "Behandling finnes ikke for behandlingId=$behandlingId",
+                             httpStatus = HttpStatus.BAD_REQUEST)
+    }
```

Comment:
Hele denne kan erstatter av behnadlingsrepository.findByIdOrThrow(behandlingId)

Reply:
Nei, jeg vil kaste en exception med tilpasset melding. Hvis findByIdOrThrow brukes, kaster det IllegalStateException.

Reply:
Dette er jo en feil som ikke skal kunne oppstå, å bruke ekstra plass på å lage en bedre feilmelding for noe som ikke skal kunne skje gir jo ikke mening.

Reply:
ok, fjernet metoden og brukte findByIdOrThrow(behandlingId) i stedet.